Use headings for configs in Configuration guide [ci-skip]

p8 opened the pull request:
### Summary

The current configurations in the Configuration guide are described
using unordered lists. This can be error prone to[ formatting errors](https://github.com/rails/rails/pull/40921)
and doesn't allow deeplinking to configuration options.

If we use headings instead, we can deeplink to these options from other guides.

p8 commented:
@jonathanhefner I thought you might like this change 😄 . It's a big diff though 😬

ceritium commented on guides/source/configuring.md:
```diff
@@ -1035,97 +1494,144 @@ You can find more detailed configuration options in the
+Accepts an array of strings regarded as web image content types in which variants can be processed without being converted to the fallback PNG format. If you want to use `WebP` or `AVIF` variants in your application you can add `image/webp` or `image/avif` to this array. The default is `%w(image/png image/jpeg image/jpg image/gif)`.
 
-* `config.active_storage.content_types_to_serve_as_binary` accepts an array of strings indicating the content types that Active Storage will always serve as an attachment, rather than inline. The default is `%w(text/html
+#### config.active_storage.web_image_content_types
```
This section is duplicated, keep the first one, it includes the AVIF variant.

p8 replied:
Thanks!

ceritium commented:
I think that this approach is better; it's easier to read, deep linkable, and less error-prone to formating errors.

I discovered that you also changed some paragraphs. I think that those changes would be better in a different commit or even a different PR. It will simplify the review process.

jonathanhefner commented on guides/source/configuring.md:
```diff
@@ -1615,15 +2117,25 @@ Initialization events
+
+Run directly after the initialization of the application, after the application initializers in `config/initializers` are run.
 
 To define an event for these hooks, use the block syntax within a `Rails::Application`, `Rails::Railtie` or `Rails::Engine` subclass:
```
This paragraph now appears to fall under the "after_initialize" subheading (whereas it was previously distinct from the "after_initialize" list item).  I'm not sure how to address that. :thinking:  Perhaps this section's list items should remain as list items for now.

p8 replied:
👍

p8 replied:
I've left this out for now.

jonathanhefner commented on guides/source/configuring.md:
```diff
@@ -1671,108 +2183,209 @@ Because `Rails::Application` inherits from `Rails::Railtie` (indirectly), you ca
 
 Below is a comprehensive list of all the initializers found in Rails in the order that they are defined (and therefore run in, unless otherwise stated).
```
I'm hesitant to convert this section's list items because it is intended to be an ordered list.  I feel like converting the items to subheadings somewhat diminishes that context.  What do you think?

p8 replied:
I've left it out for now.

jonathanhefner commented on guides/source/configuring.md:
```diff
@@ -54,149 +54,253 @@ NOTE: If you need to apply configuration directly to a class, use a [lazy load h
 These configuration methods are to be called on a `Rails::Railtie` object, such as a subclass of `Rails::Engine` or `Rails::Application`.
 
-* `config.after_initialize` takes a block which will be run _after_ Rails has finished initializing the application. That includes the initialization of the framework itself, engines, and all the application's initializers in `config/initializers`. Note that this block _will_ be run for rake tasks. Useful for configuring values set up by other initializers:
+#### config.after_initialize
```
What do you think about retaining the backticks?  e.g.

```suggestion
#### `config.after_initialize`
```

p8 replied:
Sounds good!

p8 replied:
I've added the backticks.

jonathanhefner commented on guides/source/configuring.md:
```diff
@@ -227,86 +331,134 @@ The full set of methods that can be used in this block are as follows:
+
+Converts HEAD requests to GET requests and serves them as so.
+
+#### Adding Custom Middleware
```
I think backticks on the subheadings above this one might help differentiate this subheading.

p8 replied:
Yes, good point.

jonathanhefner reviewed (commented):
Excellent job! :muscle:

p8 commented on guides/source/configuring.md:
```diff
@@ -396,637 +560,932 @@ in controllers and views. This defaults to `false`.
+#### config.active_record.dump_schemas
+
+Controls which database schemas will be dumped when calling `db:schema:dump`.
+The options are:
+
+* `:schema_search_path` (the default) which dumps any schemas listed in `schema_search_path`
+* `:all` which always dumps all schemas regardless of the `schema_search_path`
   or a string of comma separated schemas.
```
I'll leave it out for now.

p8 committed:
Use headings for configs in Configuration guide [ci-skip]

The current configurations in the Configuration guide are described
using unordered lists. This can be error prone to formatting errors and
doesn't allow deeplinking to configuration options.

If we use headings instead we can deeplink to these options from other guides.

p8 commented:
I've made the changes @jonathanhefner.

jonathanhefner reviewed (commented):
A couple more things I just noticed.

p8 commented on guides/source/configuring.md:
```diff
@@ -396,637 +560,930 @@ in controllers and views. This defaults to `false`.
+cache. It accepts an `Array` of strings, representing the table names, or
+regular expressions.
 
 The MySQL adapter adds one additional configuration option:
```
Thanks for spotting this! It also applies to the Schema Dumper.
Adding headings could make it look like this:
<img width="691" alt="image" src="https://user-images.githubusercontent.com/28561/131890185-8f6f2da5-0545-4cd0-9825-5eee675d01ad.png">

Mentioning the database could make it look like this (I slightly prefer this):
<img width="695" alt="image" src="https://user-images.githubusercontent.com/28561/131891754-bb7e56b1-8972-414d-b0db-563e6808559a.png">

jonathanhefner replied:
I think they both look fine, so I don't have a strong preference.  If we go with the first option though, it might be good to move those sections to just above [Configuring a Database](https://guides.rubyonrails.org/v6.1/configuring.html#configuring-a-database), to preserve the flow of "Configuring Active ..." sections.

p8 replied:
I went with the second option, to keep things simple.

p8 commented on guides/source/configuring.md:
```diff
@@ -396,637 +560,930 @@ in controllers and views. This defaults to `false`.
+initializer. You should run `bin/rails db:migrate` to rebuild your schema.rb
+if you change this.
 
 The schema dumper adds two additional configuration options:
```
Ah, yes :)

p8 replied:
This line has now been removed.

jonathanhefner commented:
Thank you, @p8! :tada: :tada:

p8 commented:
Great! Thanks @jonathanhefner and @ceritium for the taking time to review!
